### PR TITLE
feat: hover for ast viewing

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
This PR adds in support on the VS Code side, to be able to display the closest Generic AST node upon hover.

Companion to https://github.com/returntocorp/semgrep/pull/8142

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Ran tests locally (VSCode tests cannot run in CI)
- [] Documentation is up-to-date
- [] A changelog entry was for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by ajbt200128 — CHANGES_REQUESTED** [review]
Sorry if I was confusing when we talked about this, seeing the code while reviewing it makes it a lot clearer to me what you were asking. So there's two ways to provide hover, the first is just write one in the extension, push it like you did, and put in package.json that there's a hover provider. The 2nd is that you can add it to the lsp like [here](https://github.com/microsoft/vscode-extension-samples/tree/dbaeumer/lsp-sample-hover/lsp-sample). Note how there's nothing special on the client/extension side (i.e. no hover provider registered, no capabilities hover in the package.json), and it's all implemented on the server side. It looks like you ended up doing kind of both approaches, the SemgrepHoverProvider, and registerHoverProvider stuff ends up being redundant. This PR should just be modifying the settings to add doHover, and nothing else. My guess is that when you did just that, nothing happened, and what's missing is that on the LS side, near the bottom of `LS.ml`, you'll see a list of capabilities, and that's where you would add hover (I'll add some more comments if this explanation isn't clear).

TL;DR: This PR should just add the doHover setting, and on the LS side add hover to the list of capabilities, and everything should still work

File: package.json [hunk 1]
```diff
@@ -159,6 +162,11 @@
           "default": true,
           "description": "Only scan lines changed since the last commit"
         },
+        "semgrep.doHover": {
+          "type": "boolean",
+          "default": false,
+          "description": "Enable hovering for AST node viewing (requires restart)"
+        },
```
Comment (ajbt200128) [inline package.json:169]: Keep this, everything else is redundant

File: src/searchQuickPick.ts [hunk 2]
```diff
@@ -161,15 +161,15 @@ interface InputBoxParameters {
...
+  static async run(start: InputStep) {
     const input = new MultiStepInput();
     return input.stepThrough(start);
   }
 
   private current?: QuickInput;
   private steps: InputStep[] = [];
 
-  private async stepThrough<T>(start: InputStep) {
+  private async stepThrough(start: InputStep) {
```
Comment (ajbt200128) [inline src/searchQuickPick.ts:172]: Are these changes needed? Ok w/keeping them if it makes pre-commit/the compiler happy

**Comment by brandonspark** [comment]
@ajbt200128 tested that it works now, thank you!

for some reason pre-commit hates me, though. dunno why, i changed barely anything in this repo

**Review by ajbt200128 — APPROVED** [review]
Nice
